test/integration/test-submit-custom-header.js

```diff
@@ -31,17 +44,15 @@ var server = http.createServer(function(req, res) {
 
   form
     .on('field', function(name, value) {
-      fieldsPassed--;
-      var field = FIELDS[name];
-      assert.ok(field);
-      assert.strictEqual(value, ''+field);
+      var field = FIELDS.shift();
+      assert.strictEqual(name, field.name);
+      assert.strictEqual(value, field.value+'');
     })
     .on('file', function(name, file) {
-      fieldsPassed--;
-      var field = FIELDS[name];
-      assert.ok(field);
-      assert.strictEqual(file.name, path.basename(field.path));
-      assert.strictEqual(file.type, mime.lookup(file.name));
+      var field = FIELDS.shift();
```

Comment:
Yep. Build was right. Parsing could take longer for some things.

```
AssertionError: 'remote_file' === 'my_file'
    at IncomingForm.<anonymous> (/home/travis/build/felixge/node-form-data/test/integration/test-submit-custom-header.js:53:14)
    at emitTwo (events.js:87:13)
    at IncomingForm.emit (events.js:172:7)
    at /home/travis/build/felixge/node-form-data/node_modules/formidable/lib/incoming_form.js:229:12
    at WriteStream.<anonymous> (/home/travis/build/felixge/node-form-data/node_modules/formidable/lib/file.js:70:5)
    at WriteStream.g (events.js:260:16)
    at emitNone (events.js:72:20)
    at WriteStream.emit (events.js:166:7)
    at finishMaybe (_stream_writable.js:468:14)
    at afterWrite (_stream_writable.js:347:3)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
iojs /home/travis/build/felixge/node-form-data/test/integration/test-submit-custom-header.js
```

Let's make it hash instead of array, with name as the key, to avoid failing for nothing builds.